libraries/chain/util/smt_token.cpp

```diff
@@ -68,6 +69,262 @@ fc::optional< time_point_sec > last_emission_time( const database& db, const ass
    return {};
 }
 
+namespace generation_unit {
+
+bool is_founder( const unit_target_type& unit_target )
+{
+   return unit_target != SMT_DESTINATION_FROM && unit_target != SMT_DESTINATION_FROM_VESTING;
+}
+
+bool is_contributor( const unit_target_type& unit_target )
+{
+   return !is_founder( unit_target );
+}
+
+bool is_vesting( const unit_target_type& name )
+{
+   if ( name == SMT_DESTINATION_FROM_VESTING )
+      return true;
+   if ( name == SMT_DESTINATION_VESTING )
+      return true;
+
+   return false;
+}
+
+account_name_type get_account( const unit_target_type& unit_target )
+{
+   FC_ASSERT( is_valid_account_name( unit_target ) );
```

Comment:
Won't this fail on routes prefixed with `$`, which are not valid in account names?

I would also not name this `get_account` because it looks similar to `database::get_account`.

Reply:
This is not implemented yet, it is there so I can add `unit_target_type` and keep it compiling.

Reply:
This will (when implemented), either return `unit_target_type` as an `account_name_type`, e.g. "alice". Or extract the `account_name_type` from a special destination that is an account, e.g "$alice.vesting".